tests/test_pythonpackage.py

```diff
@@ -0,0 +1,138 @@
+
+import os
+import shutil
+import sys
+import subprocess
+import tempfile
+
+from pythonforandroid.pythonpackage import (
+    get_package_dependencies,
+    get_dep_names_of_package,
+    get_package_name,
+    _get_system_python_executable,
+)
+
+
+def test_get_package_name():
+    assert get_package_name("PiLLOw") == "Pillow"
+
+
+def test_get_package_dependencies():
+    # Check that jinja2 is returned as direct dep:
+    assert len([dep for dep in get_package_dependencies("python-for-android")
+                if "jinja2" in dep]) > 0
+    # Check that MarkupSafe is returned as indirect dep of jinja2:
+    assert [
+        dep for dep in get_package_dependencies(
+            "python-for-android", recursive=True
+        )
+        if "MarkupSafe" in dep
+    ]
+    # Check that colorama is returned without the install condition when
+    # just getting the names:
+    assert "colorama" in get_dep_names_of_package("python-for-android")
+
+
+def test__get_system_python_executable():
```

Comment:
I would definitely split this test in 2-3 parts.
So here I would make it a test class instead where you add your warning comment as a class docstring.
1. First test would stop here: `assert pyversion.strip() == sys.version.strip()`
2. Second test would start in `test_dir = tempfile.mkdtemp()` and stop in `assert os.path.normpath(sys_python_path) == os.path.normpath(pybin)`
3. Third test would probably be the Python3 dedicated one with a conditional/explicit skipif or something like that

Reply:
this is a test class now, so I'll resolve this :+1: